kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java

```diff
@@ -797,7 +654,7 @@ protected Optional<LogSegment> getLogSegmentAtOrBeforeVersion(
     }
 
     versionToLoadOpt
-        .filter(v -> v < newVersion)
+        .filter(v -> v != newVersion)
```

Comment:
why this change?

Reply:
Basically same reason as below. We want to throw an error if the version provided is after the latest version found.